lightwood/api/data_source.py

```diff
@@ -310,11 +316,16 @@ def _make_column_encoder(self, encoder_class, encoder_attrs=None, is_target=Fals
         return encoder_instance
 
     def _prepare_column_encoder(self, config, is_target=False, training_data=None):
-        column_data = self.get_column_original_data(config['name'])
-        encoder_class = config.get(
-            'encoder_class',
-            self._lookup_encoder_class(config['type'], is_target)
-        )
+        column_data = self.get_column_original_data(config['name'], is_array=config.get('historical') or config.get('previous'))
```

Comment:
Ok, so, the big problem here is that this will *remove* some behavior that we previously had, name encoding the `historical*` and `previous*` columns with a timeseries encoder *and* at the same time still waste the time of having them reshaped to arrays in mindsdb_native.

While I agree side by side encoding might be good, if we have something like:

```
order_col -> [1,2,3,4,5]
prev_target -> [0,1,4,9,16]
```

It's unclear to me why we couldn't just use the array representation of `prev_target` when encoding it along side `order_col`, i.e. apply the timeseries encoder to:

```
[[1,2,3,4,5]
 [0,1,4,9,16]] 
````

It might be that we want 3 different concepts here that are being confused:

1. A `array/timeseries auto-encoder context column` that contains a single value for each row and gets feed into the RNNEncoder used for another column
2. A `historical` column (like the current `historical*` and `previous*` columns), this is a column that gets encoded as an array/timeseries, using the RNNEncoder
3. An ability for the RNNEncoder and other ts/array encoders to auto encode two arrays combined (e.g. a `historical*` and an order by column combined)

It seems to me that this PR is implementing `1.` but in an inefficient way that breaks `2.`, though maybe I am wrong.